services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -236,16 +237,29 @@ async def _wait_until_registry_responsive(app: FastAPI) -> None:
         await _wait_until_registry_responsive(app)
 
 
+async def _list_all_services_task(app: FastAPI) -> None:
+    with log_context(_logger, logging.INFO, msg="Listing all services"):
+        await list_services(app, ServiceType.ALL)
+
+
 def setup(app: FastAPI) -> None:
     async def on_startup() -> None:
         cache = Cache(Cache.MEMORY)
         assert isinstance(cache, SimpleMemoryCache)  # nosec
         app.state.registry_cache_memory = cache
         await _setup_registry(app)
+        app_settings = get_application_settings(app)
+        app.state.auto_cache_task = None
+        if app_settings.DIRECTOR_REGISTRY_CACHING:
+            app.state.auto_cache_task = await start_periodic_task(
+                _list_all_services_task,
+                interval=app_settings.DIRECTOR_REGISTRY_CACHING_TTL / 2,
+                task_name="director-auto-cache-task",
```

Comment:
THOUGHT: imaging listing the tasks in an event loop. what would you like to see in the name listing? 

Since `servicelib.background_task.start_periodic_task` does this uniformly, I propose that instead of `task_name` to have `task_suffix` such that part of the name already adds distinctive info about the task e.g. the caller module or some distinctive properties.  This way, if we list tasks in a module we could get something like

- `servicelib.background_periodic_task/simcore_service_director.registry_proxy.on_startup/director-auto-cache-task`
- `servicelib.background_periodic_task/simcore_service_payment.services.on_startup/payment-checker`
- etc ..

This is the approach used in https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile where the caller can control prefix and suffix but not the entire name of the temporary file

![image](https://github.com/user-attachments/assets/36f872e5-d229-403c-a2fb-ee8f807609d9)

Reply:
that is a good idea. but as said. this will not belong to this PR.